src/main/java/io/reactivex/Flowable.java

```diff
@@ -8740,6 +8740,13 @@ public final Completable concatMapCompletableDelayError(Function<? super T, ? ex
      * <p>
      * Note that the operator always retains the latest item from upstream regardless of the comparison result
      * and uses it in the next comparison with the next upstream item.
+     * <p>
+     * Note that if element type {@code T} in the flow is mutable, the comparison of the previous and current
+     * item may yield unexpected results if the items are mutated externally. Common cases are mutable
+     * {@code CharSequence}s or {@code List}s where the objects will actually have the same
+     * references when they are modified and {@code distinctUntilChanged} will evaluate subsequent items as same.
+     * To avoid such situation, it is recommended that mutable data is converted to an immutable one,
+     * for example using `map(CharSequence::toString)` or `map(Collections::unmodifiableList)`.
```

Comment:
The best would be `ImmutableList::of` but we can't depend on external collections libraries. 

Perhaps we can have both by `list -> Collections.unmodifiableList(new ArrayList<>(list))`. This will make a copy of the list but also prevent it from being modified later in the pipeline. Unfortunately, we can't do much about the element mutability, which also affects `equals`.

Reply:
As discussed, I think we should change `map(Collections::unmodifiableList)` to `map(list -> Collections.unmodifiableList(new ArrayList<>(list))`.  @akarnokd Let me know if we should go forward with that and I'll create a PR.

Reply:
Yes, please modify the text as my suggestion.